### ISSUE TITLE
Pasting with (remapped) Ctrl+V does not work on Mac

### ISSUE THREAD

**Issue by aberres** [jupyter/notebook#4138]
I remapeed copy and paste shortcuts from `Cmd` to `Ctrl` as meta key (see screenshot below).

While this works nicely overall in Chrome or Safari (not in Firefox, but this is a different story) somehow Jupyter Notebook cannot deal with this. Instead of pasting content when pressing `Ctrl+V` the cursorjust  jumps to the end of the input field.
Pasting via the menus is working fine though.

Copying content with `Ctrl+C` is working fine as well.

The behavior can be seen in all Jupyter input fields. Even in the Jupyter lab settings.

I recently upgraded to Mac Os Mojave.
I am not sure if it is the OS upgrade or a jupyter upgrade, but this is something which worked "before".

```
Server Information:
You are using Jupyter notebook.

The version of the notebook server is: 5.6.0
The server is running on this version of Python:
Python 3.7.0rc1 (v3.7.0rc1:dfad352267, Jun 12 2018, 01:00:10) 
[Clang 6.0 (clang-600.0.57)]
```

![screenshot 2018-10-25 at 09 16 18](https://user-images.githubusercontent.com/20811121/47482372-aef62280-d836-11e8-9c08-ee336b895975.png)

**Comment by tfabris** [comment]
I don't use Jupyter Interactive Notebook, but I have seen that this bug is recently happening in both Chrome and Safari browsers on the Mac. I wonder if all of these programs are using some kind of the same underlying text editing engine, which is updating for each of these apps at different times.

I have noticed that when I press the (remapped) ctrl-V that the cursor jumps to the end of the input field instead of pasting, but, once that it has jumped there, then, a second press of ctrl-V will paste it there.

**Comment by tfabris** [comment]
> My guess is that this not related to Chrome or the OS, but maybe to a change in codemirror, or whatever js component is being used for code editing by Jupyter/Github.

No, it's definitely not related to Jupyter or Codemirror, because I have the same problem and I don't use either of those things. Also the same thing happens in the Safari browser on the Mac.

**Comment by clungu** [comment]
I am experiencing a similar problem. I too have remapped Cmd+V to Ctrl+V on a Mac Mojave. The behaviour is as described by @habemus-papadum (i.e. cursor jumps to the end of cell). I've looked over all the enabled extensions but none seem to use Ctrl+V. 

Would really appreciate a hint from someone, on how to mitigate this!

Thank you!

**Comment by williamevanl** [comment]
Same problem here.

**Comment by csghuser** [comment]
I have this issue with some text entry fields in chrome, but I've not tried with Jupyter. 

It appears that I can work around the issue with command+shift+v in some cases.

**Comment by Youngl41** [comment]
Same problem here

**Comment by ohke** [comment]
I faced the same problem.
Although it was unwilling, I avoided it by overriding only Chrome remapping...
<img src="https://user-images.githubusercontent.com/12320678/62913790-7a088b80-bdc8-11e9-87cf-236661fc5c76.png" width="300px">

**Comment by hdoggiedawg** [comment]
Same here.  Ctrl+V acts like in Emacs (page down). Ctrl+C doesn't do anything but make the Edit menu flash in the top bar (Finder).  I can right-click Copy and Paste ok.

**Comment by jcrben** [comment]
I reported this at https://bugs.chromium.org/p/chromium/issues/detail?id=1064501

It seems like it would be [component:Blink>Editing>Paste os=Mac](https://bugs.chromium.org/p/chromium/issues/list?q=component%3ABlink%3EEditing%3EPaste%20os%3DMac&can=2)

**Comment by tolland** [comment]
So this appears to be because the Ctrl-v is defined in `/System/Library/Frameworks/AppKit.framework/Resources/StandardKeyBinding.dict` 

```
{
...
  "^t" => "transpose:"
  "^v" => "pageDown:"
...
}
```
which is what is causing the jumping to the end of the page behaviour. There is a[ doc here](http://www.hcs.harvard.edu/~jrus/site/cocoa-text.html) which suggests that modifying:   
    /Library/KeyBindings/DefaultKeyBinding.dict
   ~/Library/KeyBindings/DefaultKeyBinding.dict

can be used to override the mappings, but this is not working for me on Mojave 10.14.6

I tried
```
{
   "^v" => "noop:";
}
```

and various combinations of that, but no luck.

**Comment by Alexsey** [comment]
It worked! Mac OS 10.15.5, tested not in jupyter but in Firefox, but the issue is definitely the same

First, it didn't work for me as well and the problem appears to be in the but of Finder or default text editor of Mac OS: when I was creating a file I was using Finder (xtraFinder) to create a text file. It was created with `.txt` extension that I deleted and edited the file with the default text editor - add
```
{
   "^v" = "noop:";
}
```
and restart the computer

And nothing happened :(

Then I go in Finder to the file - it was good. !BUT! When I go to the folder in terminal and do `ls` it showed that file has `.txt` extension!

So the right way to go is to create a file and content through the terminal (e.g. touch + nano), restart the computer and enjoy the results :)

**Comment by habemus-papadum** [comment]
amazing!  This issues had been puzzling people for so long, a simple upvote seems too lame.

**Comment by sfc-gh-sgou** [comment]
tetstetstetstetstetstetstetstets
tets

**Comment by santileortiz** [comment]
This fixed my problem in other websites, but not in places that use CodeMirror. I think it's related to this issue https://github.com/codemirror/CodeMirror/issues/5848 from CodeMirror.

**Comment by inouiw** [comment]
For me control-c and control-v work normally in codemirror.
I have no shortcuts defined but I have switched the control and command modifier keys. See screenshot. 

<img width="667" alt="image" src="https://user-images.githubusercontent.com/12761049/99309392-a461b800-2859-11eb-90bb-923c074af852.png">

**Comment by czdleaf** [comment]
Adding `"^v" = "noop:";` in `~/Library/KeyBindings/DefaultKeyBinding.dict` does solve the issue in some scenarios (for instance, when you type in Google searching text box). However, it does not affect Jupyter Notebook. When I use `ctrl-v` the cursor still jumps to the end of input field.

So here is the reason. The input field in Jupyter Notebook uses an external tool called CodeMirror, which defines some common shortcuts consistent with my Mac system. Jupyter Notebook has little control over CodeMirror, nor can it modify shortcuts defined in CodeMirror. So I just run below in the first code box in Jupyter Notebook as suggested in this link https://github.com/codemirror/codemirror5/issues/5848

```
%%js
let cm = document.querySelector(".CodeMirror");
if (cm) delete cm.CodeMirror.constructor.keyMap.emacsy["Ctrl-V"];   // delete CodeMirror's key binding of ctrl-v, so it falls back to the default action of ctrl-v which is "paste" in my system
if (cm) cm.CodeMirror.constructor.keyMap.emacsy["Ctrl-A"] = "selectAll";   // let ctrl-a be selectAll
if (cm) cm.CodeMirror.constructor.keyMap.emacsy["Ctrl-Z"] = "undo";
```

To apply `ctrl-v` in all applications, you may also want to change System Setting -> Keyboard -> Keyboard Shortcuts -> App Shortcuts and add (this alone won't work for Jupyter Notebook)
```
Copy ^C
Undo ^Z
Undo Paste ^Z
Undo Typing ^Z
Cut ^X
Paste ^V
Select All ^A
Save ^S
Find... ^F
```

**Comment by nebular** [comment]
i also have mac keys swapped and faced this on CodeMirror 6 - CTRL-V does Page Down. If I delete the mapping through the methods suggested here nothing happens. I could only make it work this way:

```
keymap.of([
	{ 
		key: "Mod-v", 
		run: view => { navigator.clipboard.readText().then(text => view.dispatch(view.state.replaceSelection(text))); return true; } 
	}
])

```

**Comment by hdoggiedawg** [comment]
Where are you entering this?  In the Jupyter cell running javascript, or in ~/Library/KeyBindings/DefaultKeyBinding.dict?
